crates/walrus-service/src/node.rs

```diff
@@ -1714,16 +1717,42 @@ impl StorageNodeInner {
         self.node_capability
     }
 
-    pub(crate) fn owned_shards(&self) -> Vec<ShardIndex> {
-        self.committee_service
-            .active_committees()
-            .current_committee()
-            .shards_for_node_public_key(self.public_key())
-            .to_vec()
+    pub(crate) fn owned_shards(&self, epoch: Epoch) -> Vec<ShardIndex> {
```

Comment:
Suggestion/nit: In many places we still use the previous behavior, namely looking for the currently owned shards. Could we keep the existing function under the name `currently_owned_shards` or similar? Same for the `is_stored_at_all_shards` function.

Reply:
Yeah, I added `owned_shards_at_<latest_>epoch` and `is_stored_at_all_shards_at_<latest_>epoch`. I feel `current` is a bit relative since it can be the current latest epoch or the current epoch the event is from.